databricks/koalas/series.py

```diff
@@ -4962,6 +4962,72 @@ def tail(self, n=5):
         """
         return first_series(self.to_frame().tail(n=n))
 
+    def product(self, min_count=0):
+        """
+        Return the product of the values.
+
+        .. note:: unlike pandas', Koalas' emulates product by ``exp(sum(log(...)))``
+            trick. Therefore, it only works for positive numbers.
+
+        .. note:: the result can be slightly difference from pandas since the rounding error.
+            In this case, use `math.isclose` to check whether the result is close to pandas' or not.
+
+             >>> result_pandas = pd.Series([10, np.nan, 30, np.nan, 50]).prod()
+             >>> result_pandas
+             15000.0
+
+             >>> result_koalas = ks.Series([10, np.nan, 30, np.nan, 50]).prod()
+             >>> result_koalas
+             15000.000000000004
+
+             >>> from math import isclose
+             >>> isclose(result_pandas, result_koalas)
+             True
+
+        Parameters
+        ----------
+        min_count : int, default 0
+            The required number of valid values to perform the operation. If fewer than
+            ``min_count`` non-NA values are present the result will be NA.
+
+        Examples
+        --------
+        >>> ks.Series([1, 2, 3, 4, 5]).prod()
+        120
+
+        By default, the product of an empty or all-NA Series is ``1``
+
+        >>> ks.Series([]).prod()
+        1.0
+
+        This can be controlled with the ``min_count`` parameter
+
+        >>> ks.Series([]).prod(min_count=1)
+        nan
+        """
+        data_type = self.spark.data_type
+        if isinstance(data_type, StringType):
+            raise TypeError("can't multiply sequence by non-int of type 'str'")
```

Comment:
What if the other data types, such as `BooleanType`, `DateType`, or `TimestampType`?

Reply:
I'll address it. Thanks!